elasticapm/transport/http.py

```diff
@@ -124,14 +137,18 @@ def get_config(self, current_version=None, keys=None):
             try:
                 max_age = int(next(re.finditer(r"max-age=(\d+)", response.headers["Cache-Control"])).groups()[0])
             except StopIteration:
-                logger.debug("Could not parse Cache-Control header: %s", response["Cache-Control"])
+                logger.debug("Could not parse Cache-Control header: %s", response.headers["Cache-Control"])
         if response.status == 304:
             # config is unchanged, return
             logger.debug("Configuration unchanged")
             return current_version, None, max_age
         elif response.status >= 400:
             return None, None, max_age
 
+        if not body:
+            logger.debug("APM Server answered with empty body and status code %s", response.status)
+            return current_version, None, max_age
```

Comment:
I don't see in this PR what the caller(s) of this function do with the body, if they handle `None` properly then this looks good to me (or: if this would return `{}` instead maybe the caller logic would be simpler?)

I am also not sure if you should return `current_version` or `response.headers.get("Etag")`... I think what you did makes sense, but I don't understand yet how this scenario can happen in apm-server :/

Reply:
Yes, the caller understands `None`.

As for version, I chose `current_version` because a new version with empty payload does not make much sense IMO, and I don't think we'd handle that properly (we do handle the config going away completely, but then it doesn't have a version either)